src/io/dataset_loader.cpp

```diff
@@ -615,10 +615,38 @@ Dataset* DatasetLoader::LoadFromBinFile(const char* data_filename, const char* b
   return dataset.release();
 }
 
+// To help verify whether sample data is changed when using different language bindings.
+static void DumpSampleData(const std::string& sample_filename, double** sample_values,
```

Comment:
I'm not sure whether this method is necessary to be included in the master branch. Is it for debugging? It seems that all calls to `ConstructFromSampleData` in the project will pass an empty `dump_filename`, so this function is not called in the source code actually. And no API is provided for users to access this method.

Reply:
Yes, this is for debugging. I suggest we find some acceptable way to keep these debugging code. It helped a lot to find problems when we develop this feature.

Reply:
BTW, c api exposes the function `LGBM_DatasetDumpText` for debugging. How about we add an option (say `dump_sample`) when creating Dataset in Python API?

Reply:
I think we'd better remove these debug code, which will increase maintenance effort. Unlike dumping a whole `Dataset`, dumping the sampling part used by construction is seldom needed. And I believe we can manually do so whenever necessary in development.
Thanks again for your contribution.

Reply:
OK. I'll remove those debugging code this week.

Reply:
@shiyu1994 Debugging code removed now. Please let me know if there's anything that should be improved.